### PR TITLE
[flash_ctrl] Fix up DIF and doc for programming the number of words

### PR DESCRIPTION
**Pull request by a-will**

### CODE REVIEW

**Comment by a-will** [comment]
CC @davebwil

File: hw/ip/flash_ctrl/data/flash_ctrl.hjson.tpl [hunk 1]
```diff
@@ -527,7 +527,7 @@
           },
           { bits: "27:16",
             name: "NUM",
-            desc: "Number of bus words the flash operation should read or program.",
+            desc: "One fewer than the number of bus words the flash operation should read or program.",
```
Comment (tjaychen) [inline hw/ip/flash_ctrl/data/flash_ctrl.hjson.tpl:530]: thanks! could you add an example just in case?
like..if you intend to program 10, write 9 here.

**Review by tjaychen — APPROVED** [review]

File: hw/ip/flash_ctrl/data/flash_ctrl.hjson.tpl (hunk 1 above)
Comment (a-will) [reply hw/ip/flash_ctrl/data/flash_ctrl.hjson.tpl]: Done :)

**Review by moidx — APPROVED** [review]
CC: @alphan

**Review by alphan — APPROVED** [review]
Just wanted to mention that there is an item for this in #3353. Should that be removed if we plan to keep this behavior?

**Comment by tjaychen** [comment]
yeah good point i will remove it.

**Comment by a-will** [comment]
Can I get a committer to merge this PR? Thanks! :)

**Pull request merged**